src/mongoose_types.erl

```diff
@@ -0,0 +1,4 @@
+-module(mongoose_types).
```

Comment:
I'm really against introducing a new module just for this. And it is used only on `ejabberd_sm`, so this type can just be defined there.